src/pip/_internal/network/session.py

```diff
@@ -76,136 +70,275 @@
+    themselves as well: the '===' or "arbitrary" operator was
+    specifically intended as an "escape hatch", for version strings that cannot be
+    made to conform to standard Python version strings
+    (https://packaging.python.org/en/latest/specifications/version-specifiers/#arbitrary-equality).
+
+    Environment markers can therefore be viewed as a language for codebases to
+    communicate their build process and requirements
+    (https://packaging.python.org/en/latest/flow/#the-packaging-flow) to a generic
+    resolver. Spack specs
+    (https://spack.readthedocs.io/en/latest/spec_syntax.html#sec-specs)
+    are very similar in spirit, and arose precisely to codify the recursive
+    and conditional relationships across multi-language software stacks.
+
+    Extending this concept, we could consider the *user agent id* as a language for
+    resolvers to communicate their requirements to external services and
+    execution environments. For example, clients for the GNU Make Jobserver protocol use
+    an environment variable to indicate how the jobserver should communicate to them,
+    and to indicate the maximum bandwidth they can tolerate for parallel execution:
+    (https://www.gnu.org/software/make/manual/html_node/Job-Slots.html).
+
+    The pants build tool specifically highlights the risk of exposing proprietary
+    information through thoughtless telemetry (https://www.pantsbuild.org/stable/docs/using-pants/anonymous-telemetry).
+    As a result, not only do they explicitly specify the information being recorded
+    (https://www.pantsbuild.org/stable/docs/using-pants/anonymous-telemetry#what-data-is-sent),
+    but they additionally incorporate anonymity as an explicit design goal
+    (https://www.pantsbuild.org/stable/docs/using-pants/anonymous-telemetry#how-we-ensure-anonymity).
+
+    Negotiating standards around useful telemetry data for PyPI began here
+    (https://github.com/pypa/pip/issues/5499), but never became a full PEP. This commit
+    (https://github.com/pypa/pip/commit/f787788a65cf7a8af8b7ef9dc13c0681d94fff5f) added
+    the output of an arbitrary subprocess execution into the string pip attaches to
+    every single HTTP request.
+
+    `PIP_USER_AGENT_USER_DATA` is mentioned in the docs once to *add* identifying info
+    to the output, in the context of a proxy server
+    (https://pip.pypa.io/en/stable/user_guide/#using-a-proxy-server).
+
+    We now introduce `PIP_TELEMETRY_USER_AGENT_ID`, which will completely overwrite the
+    string transmitted to remote hosts that pip communicates it. To maintain backwards
+    compatibility, it is disabled by default, but can be set to the empty string or any
+    other value.
+    """
 
+    @staticmethod
+    @functools.cache
+    def user_agent_id() -> str:
+        return Telemetry.calculate_user_agent_id(os.environ)
+
+    CLOBBER_USER_AGENT_ENV_VAR: ClassVar[str] = "PIP_TELEMETRY_USER_AGENT_ID"
```

Comment:
Whatever the option is names or becomes please make this a regular pip option, e.g. `--telemetry-user-agent-id`, it will therefore be accessible via command line, environmental variable (as `PIP_TELEMETRY_USER_AGENT_ID`), and configuration files.

You can see recent PRs https://github.com/pypa/pip/pull/13534 and https://github.com/pypa/pip/pull/13520 for adding new pip options.